internal/orchestrators/character/orchestrator.go

```diff
@@ -1516,6 +1521,608 @@ func convertValidationErrorsToWarnings(errors []engine.ValidationError) []Valida
+			optionMap[option.ID] = option
+		}
+
+		for _, optionID := range selection.OptionIDs {
+			option, optionExists := optionMap[optionID]
+			if !optionExists {
+				result.IsValid = false
+				result.Errors = append(result.Errors, dnd5e.ChoiceValidationError{
+					CategoryID: selection.CategoryID,
+					OptionID:   optionID,
+					Message:    fmt.Sprintf("unknown option: %s", optionID),
+					Code:       "unknown_option",
+				})
+				continue
+			}
+
+			// Check prerequisites
+			for _, prerequisite := range option.Prerequisites {
+				if !o.hasPrerequisite(draft, prerequisite) {
+					result.Warnings = append(result.Warnings, dnd5e.ChoiceValidationWarning{
+						CategoryID: selection.CategoryID,
+						OptionID:   optionID,
+						Message:    fmt.Sprintf("missing prerequisite: %s", prerequisite),
+						Code:       "missing_prerequisite",
+					})
+				}
+			}
+
+			// Check conflicts
+			for _, conflict := range option.Conflicts {
+				if o.hasConflictingChoice(draft, conflict) {
+					result.IsValid = false
+					result.Errors = append(result.Errors, dnd5e.ChoiceValidationError{
+						CategoryID: selection.CategoryID,
+						OptionID:   optionID,
+						Message:    fmt.Sprintf("conflicts with existing choice: %s", conflict),
+						Code:       "conflicting_choice",
+					})
+				}
+			}
+		}
+	}
+
+	return result, nil
+}
+
+// applyChoiceSelections applies validated selections to the draft
+func (o *Orchestrator) applyChoiceSelections(draft *dnd5e.CharacterDraft, selections []*dnd5e.ChoiceSelection) error {
+	for _, selection := range selections {
+		switch selection.CategoryID {
+		case "fighter_fighting_style":
+			draft.Choices.FightingStyles = selection.OptionIDs
+		case "wizard_cantrips":
+			draft.Choices.Cantrips = selection.OptionIDs
+		case "wizard_spells":
+			draft.Choices.Spells = selection.OptionIDs
+		case "additional_languages":
+			draft.Choices.Languages = selection.OptionIDs
+		case "tool_proficiencies":
+			draft.Choices.Tools = selection.OptionIDs
+		case "equipment_choices":
+			draft.Choices.Equipment = selection.OptionIDs
+		default:
+			return errors.InvalidArgumentf("unknown choice category: %s", selection.CategoryID)
+		}
+	}
+	return nil
+}
+
+// getAvailableChoiceCategories returns the choice categories available for a draft
+func (o *Orchestrator) getAvailableChoiceCategories(
+	ctx context.Context,
+	draft *dnd5e.CharacterDraft,
+	filterType *dnd5e.ChoiceType,
+) ([]*dnd5e.ChoiceCategory, error) {
+	var categories []*dnd5e.ChoiceCategory
+
+	// Get class details to determine available choices
+	classDetails, err := o.GetClassDetails(ctx, &GetClassDetailsInput{ClassID: draft.ClassID})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get class details")
+	}
+
+	class := classDetails.Class
+
+	// Add class-specific choices based on class ID
+	switch draft.ClassID {
+	case "fighter":
+		if filterType == nil || *filterType == dnd5e.ChoiceTypeFightingStyle {
```

Comment:
can we do this filtertype nil checks differently? do we have to dereference it? can we do that one time?